components/support/nimbus-fml/src/client/inspector.rs

```diff
@@ -104,40 +106,25 @@ impl FmlFeatureInspector {
     }
 
     fn get_semantic_errors(&self, src: &str, value: Value) -> Vec<FmlEditorError> {
-        let errors = self
+        let (merged_value, errors) = self
             .manifest
-            .get_errors(&self.feature_id, &value)
+            .merge_and_errors(&self.feature_id, &value)
             .unwrap_or_else(|e| {
                 unreachable!("Error {e:?} should be caught as FeatureValidationError")
             });
-        let mut editor_errors: Vec<_> = Vec::with_capacity(errors.len());
-        for e in errors {
-            let message = e.message;
-            let highlight = e.path.last_token().map(str::to_string);
-            let (line, col) = e.path.line_col(src);
-            let error = FmlEditorError {
-                message,
-                line: line as u32,
-                col: col as u32,
-                highlight,
-            };
-            editor_errors.push(error);
+        if !errors.is_empty() {
+            let (manifest, feature_def) = self.manifest.find_feature(&self.feature_id).unwrap();
+            let converter = ErrorConverter::new(&manifest.enum_defs, &manifest.obj_defs);
+            converter.convert_into_editor_errors(feature_def, &merged_value, src, &errors)
```

Comment:
Instantiating a type just to call a single method is a little code smelly. If we use this pattern everywhere (i.e., ErrorConverter instances are used only once) we should probably replace it with a free fn.

```suggestion
            ErrorConverter::new(&manifest.enum_defs, &manifest.obj_defs)
                .convert_into_editor_errors(feature_def, &merged_value, src, &errors)
```

Reply:
I considered a free function, but found that there were a lots of passing arguments around. We're using ErrorConverter + method in a few places, but always different methods.

I like the different structs to collect methods together, rather than one great big god object (as `FeatureManifest` was getting to be).

The other alternative is to have a struct with the `enum_defs` and `obj_defs` and pass that around, like a different spelling of `&self`.

WDYT?